pandas/core/dtypes/concat.py

```diff
@@ -604,12 +603,8 @@ def _concat_rangeindex_same_dtype(indexes):
         # Get the stop value from "next" or alternatively
         # from the last non-empty index
         stop = non_empty_indexes[-1]._stop if next is None else next
+        return RangeIndex(start, stop, step)
     else:
```

Comment:
maybe remove the else here (and just return the RangeIndex)

Reply:
ok, I personally like the else in such situations, it's more explicit to me, I guess it's a matter of taste. I'll remove it :-)